### PR TITLE
Update TerraWDIOTestDetailsReporter to reset the screenshots array after each test

### PR DESCRIPTION
**Pull request by sirishaNekkanti**
### Summary
Small fix to reset the screenshots array after every test

Also updated the docs for the TerraWDIOTestDetailsReporter to reflect the changes we made to the result file where file name should start with functional-test-details and also updated the sample results structure of the TerraWDIOTestDetailsReporter

### CODE REVIEW

**Review by yuderekyu — COMMENTED** [review]
can we add a unit test to catch this scenario?

File: tests/jest/TerraWDIOTestDetailsReporter.test.js [hunk 1]
```diff
@@ -244,6 +244,13 @@ describe.only('TerraWDIOTestDetailsReporter', () => {
       expect(typeof reporter.specHashData[params.specHash][params.title].tests).toEqual('object');
     });
   });
+  describe('test:end', () => {
+    it('test:end should reset the screenshots array ', () => {
+      const reporter = new TerraWDIOTestDetailsReporter({}, {});
+      reporter.emit('test:end', { title: 'title of the it' });
+      expect(reporter.screenshots.length).toBeGreaterThanOrEqual(0);
```
Comment (yuderekyu) [inline tests/jest/TerraWDIOTestDetailsReporter.test.js:251]: This test can pass if the array has items. Let's constrain the assertion and do an equals check on 0 instead. Do we also need a verification pull request, or integration test, for this change?

Comment (yuderekyu) [reply tests/jest/TerraWDIOTestDetailsReporter.test.js]: sorry i'm not familiar with the reporter, but on second thought, is the screenshots array actually filled by just calling `end`, or will it always be empty? if it's not filled in `end` we should populate it first before emitting `end`

Comment (sirishaNekkanti) [reply tests/jest/TerraWDIOTestDetailsReporter.test.js]: I did tested this change in orion-release-site and terra-clinical

Comment (yuderekyu) [reply tests/jest/TerraWDIOTestDetailsReporter.test.js]: I am wondering if the `screenshots` array will always be empty in this test case. Do we need to do something like below, or is screenshots initially populated in your example?
```js
reporter.screenshots = ['real world input here'];
reporter.emit('end');
expect(screenshots.length).toEqual(0);
```

Comment (sirishaNekkanti) [reply tests/jest/TerraWDIOTestDetailsReporter.test.js]: Not always empty, it is empty at the end. I added these changes https://github.com/cerner/terra-toolkit-boneyard/blob/dr-screenshots/tests/jest/TerraWDIOTestDetailsReporter.test.js#L250-L252

**Review by mjhenkes — APPROVED** [review]

**Review by StephenEsser — APPROVED** [review]

**Review by yuderekyu — APPROVED** [review]

**Review by ryanthemanuel — APPROVED** [review]

**Pull request merged**